src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1081,20 +1126,36 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
                     spawnCitizen();
                 }
             }
+
+            if (!raidHasHappened && !subscribers.isEmpty() && world != null && !world.isDaytime() && Configurations.doBarbariansSpawn && BarbarianUtils.raidThisNight(world))
+            {
+                raidLevel = this.numberOfWorkerLevels();
+                if (citizens.size() < CITIZEN_MINIMUM_FOR_RAID)
+                {
+                    raidLevel = 1;
+                }
+                BarbarianUtils.eventRaid(world, raidLevel, this);
+                raidHasHappened = true;
+            }
+
+            if (world != null && world.isDaytime())
+            {
+                raidHasHappened = false;
+            }
         }
 
         //  Tick Buildings
         for (@NotNull final AbstractBuilding building : buildings.values())
         {
             building.onWorldTick(event);
         }
-
-        if(isDay && !world.isDaytime())
+        
+        if (isDay && !world.isDaytime())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "world" is nullable here [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)